parachain/runtime/common/src/lib.rs

```diff
@@ -113,3 +112,19 @@ where
         let _ = NativeCurrency::resolve_creating(&TreasuryAccount::get(), amount);
     }
 }
+
+pub struct MaybeSetTimestamp<T>(PhantomData<T>);
+
+impl<T> OnTimestampSet<T::Moment> for MaybeSetTimestamp<T>
+where
+    T: frame_system::Config + pallet_aura::Config + pallet_sudo::Config,
+{
+	fn on_timestamp_set(moment: T::Moment) {
+        // key is not set on mainnet
+        if pallet_sudo::Pallet::<T>::key().is_none() {
```

Comment:
I'm not sure if this is the best approach but I couldn't see an obvious alternative. @sander2 what do you think?

Reply:
A bit hacky for sure but without knowing why exactly the hook fails I can't really suggest better alternatives. I guess it's fine

Reply:
This is the error: https://github.com/paritytech/substrate/blob/83bf140ca43df76ccaf422385562e33782ed43f8/frame/timestamp/src/lib.rs#L176

Reply:
Which causes the panic here: https://github.com/paritytech/substrate/blob/83bf140ca43df76ccaf422385562e33782ed43f8/frame/timestamp/src/lib.rs#L176